src/common/get-unavailable-highlight-status.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    PlatformData,
+    UnifiedResult,
+    ViewPortProperties,
+} from 'common/types/store-data/unified-data-interface';
+
+import { HighlightState } from 'common/components/cards/instance-details-footer';
+
+import { BoundingRectangle } from 'electron/platform/android/scan-results';
+
+export type GetUnavailableHighlightStatus = (
+    result: UnifiedResult,
+    platformInfo: PlatformData,
+) => HighlightState;
```

Comment:
I found it confusing that the function signature suggests it should be able to return `'unavailable' | 'visible' | 'hidden'`, but the actual behavior of all implementations is to return `'unavailable' | null`

Reply:
Since the only callsite is relying on the `'unavailable` | null` behavior, I'd suggest just updating this to return a boolean instead and rename stuff here accordingly; I think it would make the callsite, type signature, and implemenations all more clear

Reply:
I was definitely favoring having other potential statuses because I figured we'd want to mention the reason (and this is consumed by the view as is) ... I guess on second-thought it might not be a bad idea to simplify the statuses (i.e. keep them as they are with unavailable, visible, hidden) and then later make it an object providing additional information about the reason.

That would require the view not directly depending on this value in the future, which is probably a good thing as well.